src/ArticleTheme.vala

```diff
@@ -0,0 +1,109 @@
+//	This file is part of FeedReader.
+//
+//	FeedReader is free software: you can redistribute it and/or modify
+//	it under the terms of the GNU General Public License as published by
+//	the Free Software Foundation, either version 3 of the License, or
+//	(at your option) any later version.
+//
+//	FeedReader is distributed in the hope that it will be useful,
+//	but WITHOUT ANY WARRANTY; without even the implied warranty of
+//	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//	GNU General Public License for more details.
+//
+//	You should have received a copy of the GNU General Public License
+//	along with FeedReader.  If not, see <http://www.gnu.org/licenses/>.
+using Gee;
+
+public class  FeedReader.ArticleTheme {
+    private static ArrayList<HashMap<string, string>> ? themes = null;
+
+
+    public static ArrayList<HashMap> getThemes(){
+        if(themes == null){
+            // Local themes
+            themes = new ArrayList<HashMap<string, string>> ();
+            string local_dir = GLib.Environment.get_user_data_dir() + "/feedreader/themes/";
+            grabThemes(local_dir);
+            // Global themes
+            string global_dir = Constants.INSTALL_PREFIX + "/share/FeedReader/ArticleView/";
+            grabThemes(global_dir);
+        }
+        return themes;
+    }
+
+    private static HashMap<string, string> getThemeInfo (string theme_path) {
+      var themeInfo = new HashMap<string, string> ();
+      bool corrupted_theme = true;
+      string theme_name = "";
+      string author = "";
+      try {
+        Dir theme_dir = Dir.open(theme_path, 0);
+        string ? name = null;
+        while ((name = theme_dir.read_name()) != null){
+          if (name == "theme.json"){
+              corrupted_theme = false;
+              string path = Path.build_filename(theme_path, name);
```

Comment:
The code here is still not complete, i would like to check if there's an article.html & style.css file too. Otherwise the theme will be correct. But yeah it's better to use this :+1: